src/IlluminateRegistry.php

```diff
@@ -302,7 +302,7 @@ public function getRepository(string $persistentObject, string|null $persistentM
      *
      * @param class-string $className A persistent object class name.
```

Comment:
We could add `@return`  annotation, so static tools (e.g. `phpstan`) will understand that return type depends on argument

`* @return ($throwExceptionIfNotFound is true ? ObjectManager : ObjectManager|null)`